pkg/controller/jivavolume/jivavolume_controller.go

```diff
@@ -0,0 +1,709 @@
+							},
+						},
+					},
+				}).
+				WithContainerBuilders(
+					container.NewBuilder().
+						WithName("jiva-replica").
+						WithImage(getImage("OPENEBS_IO_JIVA_REPLICA_IMAGE",
+							"jiva-replica")).
+						WithPortsNew([]corev1.ContainerPort{
+							{
+								ContainerPort: 9502,
+								Protocol:      "TCP",
+							},
+							{
+								ContainerPort: 9503,
+								Protocol:      "TCP",
+							},
+							{
+								ContainerPort: 9504,
+								Protocol:      "TCP",
+							},
+						}).
+						WithCommandNew([]string{
+							"launch",
+						}).
+						WithArgumentsNew([]string{
+							"replica",
+							"--frontendIP",
+							fmt.Sprintf(svcNameFormat, cr.Name, cr.Namespace),
+							"--size",
+							fmt.Sprint(capacity),
+							"openebs",
+						}).
+						WithImagePullPolicy(corev1.PullIfNotPresent).
+						WithPrivilegedSecurityContext(&prev).
+						WithResources(&cr.Spec.ReplicaResource).
+						WithVolumeMountsNew([]corev1.VolumeMount{
+							{
+								Name:      "openebs",
+								MountPath: "/openebs",
+							},
+						}),
+				),
+		).
+		WithPVC(
+			pvc.NewBuilder().
+				WithGenerateName("openebs-jiva-replica-pvc").
+				WithNamespace(cr.Namespace).
+				WithOwnerReferenceNew([]metav1.OwnerReference{{
+					APIVersion:         cr.APIVersion,
+					BlockOwnerDeletion: &blockOwnerDeletion,
+					Controller:         &controller,
+					Kind:               cr.Kind,
+					Name:               cr.Name,
+					UID:                cr.UID,
+				},
+				}).
+				WithStorageClass(cr.Spec.ReplicaSC).
+				WithAccessModes([]corev1.PersistentVolumeAccessMode{corev1.ReadWriteOnce}).
+				WithCapacity(fmt.Sprint(capacity)),
+		).Build()
+
+	if err != nil {
+		return fmt.Errorf("failed to build statefulset object, err: %v", err)
+	}
+
+	found := &appsv1.StatefulSet{}
+	err = r.client.Get(context.TODO(), types.NamespacedName{Name: stsObj.Name, Namespace: stsObj.Namespace}, found)
+	if err != nil && errors.IsNotFound(err) {
+		// Set JivaVolume instance as the owner and controller
+		if err := controllerutil.SetControllerReference(cr, stsObj, r.scheme); err != nil {
+			return err
+		}
+
+		reqLog.V(2).Info("Creating a new Statefulset", "Statefulset.Namespace", stsObj.Namespace, "Sts.Name", stsObj.Name)
+		err = r.client.Create(context.TODO(), stsObj)
+		if err != nil {
+			return err
+		}
+		// Statefulset created successfully - don't requeue
+		return nil
+	} else if err != nil {
+		return operr.Wrapf(err, "failed to get the statefulset details: %v", stsObj.Name)
+	}
+
+	return nil
+}
+
+func updateJivaVolumeWithServiceInfo(r *ReconcileJivaVolume, cr *jv.JivaVolume, reqLog logr.Logger) error {
+	ctrlSVC := &v1.Service{}
+	if err := r.client.Get(context.TODO(),
+		types.NamespacedName{
+			Name:      cr.Name + "-jiva-ctrl-svc",
+			Namespace: cr.Namespace,
+		}, ctrlSVC); err != nil {
+		return fmt.Errorf("%s, err: %v", updateErrMsg, err)
+	}
+	cr.Spec.ISCSISpec.TargetIP = ctrlSVC.Spec.ClusterIP
```

Comment:
What if clusterIP is not set even after waiting for 1 second, we don't have a validation check for the same.

Reply:
we have Get call over this, once service is created, it will have valid serviceIP